base/refpointer.jl

```diff
@@ -142,3 +142,17 @@ getindex(b::RefArray) = b.x[b.i]
 setindex!(b::RefArray, x) = (b.x[b.i] = x; b)
 
 ###
+
+if Int === Int64
+    primitive type AddrSpacePtr{T, AS} <: Ref{T} 64 end
```

Comment:
The datatype needs to be defined in the C part, see https://github.com/JuliaLang/julia/blob/c3f6542aa3f90485f4b5fbac0486c390df7284d5/src/jltypes.c#L2233

Reply:
Do I need to register type anywhere else but in `src/builtins.c`, `src/jltypes.c` and `julia.h`?
Building the sysimage seems to fail because `Core.AddrSpacePtr` is unknown:
```
refpointer.jl
error during bootstrap:

jl_undefined_var_error at /home/tfaingna/src/julia/src/rtutils.c:130
jl_eval_global_var at /home/tfaingna/src/julia/src/interpreter.c:393 [inlined]
eval_value at /home/tfaingna/src/julia/src/interpreter.c:439
eval_stmt_value at /home/tfaingna/src/julia/src/interpreter.c:409 [inlined]
eval_body at /home/tfaingna/src/julia/src/interpreter.c:821
jl_interpret_toplevel_thunk at /home/tfaingna/src/julia/src/interpreter.c:915
top-level scope at refpointer.jl:146
jl_toplevel_eval_flex at /home/tfaingna/src/julia/src/toplevel.c:814
jl_parse_eval_all at /home/tfaingna/src/julia/src/ast.c:907
jl_load_rewrite at /home/tfaingna/src/julia/src/toplevel.c:872
include at ./boot.jl:329 [inlined]
include at ./Base.jl:15
include at ./Base.jl:17
_jl_invoke at /home/tfaingna/src/julia/src/gf.c:2144 [inlined]
jl_apply_generic at /home/tfaingna/src/julia/src/gf.c:2328
jl_apply at /home/tfaingna/src/julia/src/julia.h:1703 [inlined]
do_call at /home/tfaingna/src/julia/src/interpreter.c:369
eval_value at /home/tfaingna/src/julia/src/interpreter.c:458
eval_stmt_value at /home/tfaingna/src/julia/src/interpreter.c:409 [inlined]
eval_body at /home/tfaingna/src/julia/src/interpreter.c:803
jl_interpret_toplevel_thunk at /home/tfaingna/src/julia/src/interpreter.c:915
top-level scope at Base.jl:93
jl_toplevel_eval_flex at /home/tfaingna/src/julia/src/toplevel.c:814
jl_eval_module_expr at /home/tfaingna/src/julia/src/toplevel.c:181
jl_toplevel_eval_flex at /home/tfaingna/src/julia/src/toplevel.c:640
jl_parse_eval_all at /home/tfaingna/src/julia/src/ast.c:907
jl_load_rewrite at /home/tfaingna/src/julia/src/toplevel.c:872
include at ./boot.jl:329
_jl_invoke at /home/tfaingna/src/julia/src/gf.c:2161 [inlined]
jl_apply_generic at /home/tfaingna/src/julia/src/gf.c:2328
jl_apply at /home/tfaingna/src/julia/src/julia.h:1703 [inlined]
do_call at /home/tfaingna/src/julia/src/interpreter.c:369
eval_value at /home/tfaingna/src/julia/src/interpreter.c:458
eval_stmt_value at /home/tfaingna/src/julia/src/interpreter.c:409 [inlined]
eval_body at /home/tfaingna/src/julia/src/interpreter.c:803
jl_interpret_toplevel_thunk at /home/tfaingna/src/julia/src/interpreter.c:915
top-level scope at sysimg.jl:3
jl_toplevel_eval_flex at /home/tfaingna/src/julia/src/toplevel.c:814
jl_parse_eval_all at /home/tfaingna/src/julia/src/ast.c:907
jl_load_rewrite at /home/tfaingna/src/julia/src/toplevel.c:872
exec_program at /home/tfaingna/src/julia/ui/repl.c:35
true_main at /home/tfaingna/src/julia/ui/repl.c:108
main at /home/tfaingna/src/julia/ui/repl.c:217
__libc_start_main at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
_start at /home/tfaingna/src/julia/usr/bin/julia (unknown line)
```

(`make clean` doesn't fix this)

Reply:
Ah, I suppose I have to add it to `src/staticdata.c` and `src/dump.c` as well

Reply:
Nevermind, just noticed I forgot the `Core.` which is needed because `AddrSpacePtr` is not exported... :man_facepalming: